3-CustomWebUI/3-2-CustomBrowser/Console-Interactive-CustomWebUI/Program.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+using System;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Console_Interactive_CustomWebUI
+{
+    class Program
+    {
+        private static PublicClientApplicationOptions appConfiguration = null;
+        private static IConfiguration configuration;
+        private static string _authority;
+
+        static async Task Main(string[] args)
+        {
+            // Using appsettings.json as our configuration settings
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(System.IO.Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json");
+
+            configuration = builder.Build();
+
+            appConfiguration = configuration
+                .Get<PublicClientApplicationOptions>();
+
+            _authority = string.Concat(appConfiguration.Instance, appConfiguration.TenantId);
+
+            // Building a public client application
+            var app = PublicClientApplicationBuilder.Create(appConfiguration.ClientId)
+                                                    .WithAuthority(_authority)
+                                                    .WithRedirectUri(appConfiguration.RedirectUri)
+                                                    .Build();
+
+            string[] scopes = new[] { "user.read" };
+            AuthenticationResult result;
+
+            try
+            {
+                var accounts = await app.GetAccountsAsync();
+
+                // Try to acquire an access token from the cache. If an interaction is required, 
+                // MsalUiRequiredException will be thrown.
+                result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
+                            .ExecuteAsync();
+            }
+            catch (MsalUiRequiredException)
+            {
+                // Acquiring an access token interactively. MSAL will cache it so we can use AcquireTokenSilent
+                // on future calls.
+                result = await app.AcquireTokenInteractive(scopes)
+                            .ExecuteAsync();
+            }
+
+            string graphApiUrl = configuration.GetValue<string>("GraphApiUrl");
+            // Instantiating GraphServiceClient and using the access token acquired above.
+            var graphClient = GetGraphServiceClient(result.AccessToken, graphApiUrl);
```

Comment:
@TiagoBrenck you missed this one?

Reply:
Oh, 3-2 is actually not even ready. Thanks for pointing that

Reply:
Updated @jmprieur